---
title: "Gene Expression Prediction Challenge (expred)"
subtitle: "Starting kit"
author: "Florent Chuffart & Magali Richard"
date: "`r Sys.Date()`"
output: 
  rmarkdown::html_document:
    toc: true
    toc_float: true
    toc_depth: 3
    number_sections: false
---

```{r label="header", echo=FALSE, eval=TRUE}
knitr::opts_chunk$set(collapse=TRUE, comment = "#>", fig.width=9, fig.height=6, eval=TRUE, echo=FALSE, results="hide")
d = readRDS(file="data.rds")
```


# I. General information


The challenge provides the data.frame `d` of non tumoral and tumoral tissues described by genes expression values, histological and clinical attributes.

Some gene expression values are missing.

**The goal of the challenge** is to use statistical models (*e.g.* linear models) to predict missing gene expression values using provided gene expression values, histological and clinical attributes.

 - Biological attributes are:    `r names(d)[c(1:2,4)]`.
 - Histological attribuites are: `r names(d)[c(5,3)]`
 - Clinical attributes are:      `r names(d)[6:10]`
 - Gene columns are named:

```{r, echo=TRUE}
gs = c(
  "ATAD2", "SYCP3",                # Chromatin 
  "H19", "IGF2", "NNAT", "BLCAP",  # inprinted genes
  "BRD4", "BRDT", "NUTM1",         # Testis restricted
  "MAGEB6", "TUBA3C",              # Testis specific
  "SMYD3", "MAP3K2", "KDR",        # K methyl transferase
  "TP53", "KRAS", "BRAF",          # Tumor supressors
  "FASTKD1", "ND2", "ND3", "ND4"   # Mitochondiral activity
)
```


```{r results="verbatim", echo=TRUE}
table(is.na(d[,gs]))
```

```{r results="verbatim", echo=TRUE}
head(d[,gs])
head(d[,1:10])
dim(d)
```


# II. How to participate?

## Where to go? 

The challenge is hosted on the codalab platform:

https://competitions.codalab.org/competitions/22389?secret_key=a1edad61-8bfe-4fb4-a62e-c05ab4873411

To participate, you need to:

1- create an acount on the codalab platform

2- subscribe to the *expred* challenge

3- constitute teams (up to 4 people) and design a team coordinator on the platform

## How to download data and submit results?

[1] Go on the challenge page, in the *Learn the details* tab, in the *get_starting_kit* item and download the starting kit by clicking the *Starting Kit* button.

[2] On your local machine, unzip the just downloaded zip file *stating_kit.zip* and open *R* in the unziped strating-kit directory, (e.g. open *strating_kif.Rmd* with *RStudio*).

This folder contains:

- A data.rds file containing the data.
- A scoring program folder containing the program to compute the scoring (for you information).
- A starting_kit.Rmd corresponding to the vignette of the Challenge (all useful information can be found here).
- A submission_script.R to modify and to use to submit your predictions.

[3] In the R console launch the following command: 

`rmarkdown::render("starting_kit.Rmd")`

It procudes the file "starting_kit.html" that explains how to analyse data and produce results.

[4] Generate a missing value prediction. Please see a detailed example of submission script in the **Submission** section.

[5] Submit your prediction (zip file) in the *Participate* tab of the codalab challenge.

The metric used to evaluate prediction is the  **mean squared error** (MSE)

$$MSE = \frac{1}{n}\sum_{i=1}^n{(Y_i-\widehat{Y}_i )^2}$$



# III. Report / Evaluation

## Report

The goal will be to predict missing values in a gene expression dataset. 

You will use the codalab platform to evaluate your methods. 

The challenge will be open for a week on the platform.
You will work in teams of 3 or 4.

You will report by team. 
The report must be returned by Wednesday, April 17th, before 2pm.

This report should consist of 3 parts, with:

**A) Statistical methods used**

*For instance* :

- Descriptive statistics (actors, number of NA, pretreatment...)
- Methods: Which function for linear model? (*e.g* lm, aov, rlm, glm, other...), Which model are your main models? (ND2 ~ sex+age+tissue_status)...

**B) Critical evaluation of the progression of the score**

*For instance* :
For critical variation in your leader board score:

- Evaluation of your model (e.g. explained variance $R^2$, number of parameters...)
- What is your score on codalab?
    
**C) Discussion and biological interpretation**

*For instance* :

- Pros and cons each method  
- Biological meaning (gene expression correlations, tisuus status...)

## Evaluation

You will be evaluated on the report you provide (hability to clearly explain your approach, the methods used and the evolution of your prediction score), the codalab leader board score **WILL NOT** be used as a metric to evaluate your score. 







# IV. Submission  (a detailed example)

## Generate a missing value prediction

Please use the `submission_script.R` contained in the `starting_k` folder as a template to write, run and save your code.

```{r}
zip_filename = paste(sep="",  "results_", format(Sys.time(), format="%m_%d_%Y_%s"), ".zip")
```

In this exemple, we randomly fill missing values and generate *`r zip_filename`* submission file.

```{r echo=TRUE, results="verbatim"}
# Predict the values
pred = as.matrix(d[,gs])
pred[] = sample(na.omit(pred), nrow(pred)*ncol(pred), replace=TRUE)
pred[!is.na(as.matrix(d[,gs]))] = NA # Sparse, only submit NA values!
# Save and export the results
saveRDS(pred, "results.rds")
zip(zip_filename, "results.rds")
print(zip_filename)
```

In this exemple, for each gene, we use a simple linear to predict missing values.

```{r echo=TRUE, results="verbatim"}
# Predict the values
pred = sapply(gs, function(g, d){
  m = lm(d[[g]] ~ d$tissue)
  # m = lm(d[[g]] ~ d$age+d$sex+d$tissue+d$tissue_status+d$project)
  predict(m, d)
}, d)
pred[!is.na(as.matrix(d[,gs]))] = NA # Sparse, only submit NA values!
# Save and export the results
saveRDS(pred, "results.rds")
zip(zip_filename, "results.rds")
print(zip_filename)
```

```{r}
# In this exemple, we use a simple linear to predict all missing values.
# m = lm(as.matrix(d[,gs]) ~ d$tissue)
# summary(m)
# anova(m)
# pred = predict(m, d)
# pred[!is.na(as.matrix(d[,gs]))] = NA # Sparse, only submit NA values!
```

## Submit zip file

Submit your *`r zip_filename`* submission file in the *Participate* tab of the codalab challenge by clicking the *Submit* button.

## Evaluation

The metric used to evaluate prediction is the mean squared error (MSE) coded is the following scoring program (provided).

```{r echo=TRUE, results="verbatim"}
source("scoring_program/scoring.r")
```

The *STATUS* become : 

  - Submitted	
  - Running
  - Finished

When itâ€™s finished, score is given in the column *SCORE* and details for report could be downloaded by clicking *Download output from scoring step*.


# V. Few words about the dataset 


```{r, eval=FALSE}
m = lm(as.matrix(d[,gs]) ~ d$age+d$sex+d$tissue+d$tissue_status+d$project)
data = as.matrix(d[,gs])
data[is.na(d[,gs])] = predict(m, d)[is.na(d[,gs])]

pca = prcomp(data, scale=TRUE)
v = pca$sdev * pca$sdev
p = v / sum(v) * 100
layout(matrix(1:6,2, byrow=FALSE), respect=TRUE)
barplot(p)
i=3
j=2
plot(pca$x[,i], pca$x[,j], xlab=paste(sep="", "PC", i, "(", signif(p[i], 3), "%)"), ylab=paste(sep="",  "PC", j, "(", signif(p[j], 3), "%)"), pch=".")
i=1
j=3
plot(pca$x[,i], pca$x[,j], xlab=paste(sep="",  "PC", i, "(", signif(p[i], 3), "%)"), ylab=paste(sep="",  "PC", j, "(", signif(p[j], 3), "%)"), pch=".")
i=1
j=2
plot(pca$x[,i], pca$x[,j], xlab=paste(sep="",  "PC", i, "(", signif(p[i], 3), "%)"), ylab=paste(sep="",  "PC", j, "(", signif(p[j], 3), "%)"))
```

```{r eval=FALSE}
sapply(gs, function(g){
  boxplot(d[[g]] ~ d$tissue, las=2, main=paste(g, "~ tissue"))  
})
```





# VI. References 


[NCBI/GENE] https://www.ncbi.nlm.nih.gov/gene

[TCGA] Data are coming from the TCGA (The cancer genome atlas). 
They were extracted from several cohorts of patients. 
Here are TCGA Study Abbreviations corresponding to each cancer. 
https://gdc.cancer.gov/resources-tcga-users/tcga-code-tables/tcga-study-abbreviations

[Codalab] https://competitions.codalab.org

[R] https://cran.r-project.org

[RStudio] https://www.rstudio.com





# VII. Session Information

```{r, results="verbatim"}
sessionInfo()
```


