---
title: "Gene Expression Prediction Challenge (expred)"
subtitle: "Starting kit"
author: "Florent Chuffart & Magali Richard"
date: "`r Sys.Date()`"
output: 
  rmarkdown::html_document:
    toc: true
    toc_float: true
    toc_depth: 3
    number_sections: false
---

```{r label="header", echo=FALSE, eval=TRUE}
knitr::opts_chunk$set(collapse=TRUE, comment = "#>", fig.width=9, fig.height=6, eval=TRUE, echo=FALSE, results="hide")

gs = c(
 "ATAD2", "SYCP3",
 "BRD4", "BRDT", "NUTM1",
 "MAGEB6", "TUBA3C",
 "H19", "IGF2", "NNAT", "BLCAP",
 "SMYD3", "MAP3K2", "KDR",
 "TP53", "KRAS", "BRAF",
 "FASTKD1", "ND2", "ND3", "ND4" 
)

```


# I. General information


The challenge provides the data.frame `d` of non tumoral and tumoral tissues described by genes expression values, histological and clinical attributes.

Some gene expression values are missing.

**The goal of the challenge** is to use statistical models (*e.g.* linear models) to predict missing gene expression values using provided gene expression values, histological and clinical attributes.

 - Biological attributes are:    `r names(d)[c(1:2,4)]`.
 - Histological attribuites are: `r names(d)[c(5,3)]`
 - Clinical attributes are:      `r names(d)[6:10]`
 - Gene columns are named: `r gs`.

```{r results="verbatim", echo=TRUE}
d = readRDS(file="data.rds")
head(d)
dim(d)
```

```{r results="verbatim", echo=TRUE}
table(is.na(d[,gs]))
```

# II. How to participate ?

## Where to go? 

The challenge is hosted on the codalab platform:

https://competitions.codalab.org/competitions/21360?secret_key=1b0893b2-c695-4c9b-ae33-334ac7a056ac

To participate, you need to:

1- create an acount on the codalab platform

2- subscribe to the *expred* challenge

3- constitute teams (up to 4 people) and design a team coordinator on the platform

## How to download data and submit results?

[1] In the participate window, go in the Files tab and download the starting kit.

[2] Go to the strating-kit directory and launch in R the following command: `rmarkdown::render("starting_kit.Rmd")`

It procudes the file "starting_kit.html" that explains how to analyse data and produce results.

[3] Generate a missing value prediction.

[4] Submit in (zip file) in the Participate tab of the codalab challenge.

[5] The metric used to evaluate prediction is the  **mean squared error** (MSE)

$$MSE = \frac{1}{n}\sum_{i=1}^n{(Y_i-\widehat{Y}_i )^2}$$


Please see a detailed example of submission script in the **Submission** section



# III. Report / Evaluation

## Report

The goal will be to predict missing values in a gene expression dataset. 

You will use the codalab platform to evaluate your methods. 

The challenge will be open for a week on the platform.
You will work in teams of 3 or 4.

You will report by team. 
The report must be returned by Wednesday, April 17th, before 2pm.

This report should consist of 3 parts, with:

**A) Statistical methods used**

*For instance* :

- Descriptive statistics (actors, number of NA, pretreatment...)
- Methods: Which function for linear model? (*e.g* lm, aov, rlm, glm, other...), Which model are your main models? (ND2 ~ sex+age+tissue_status)...

**B) Critical evaluation of the progression of the score**

*For instance* :
For critical variation in your leader board score:

- Evaluation of your model (e.g. explained variance $R^2$, number of parameters...)
- What is your score on codalab?
    
**C) Discussion and biological interpretation**

*For instance* :

- Pros and cons each method  
- Biological meaning (gene expression correlations, tisuus status...)

## Evaluation

You will be evaluated on the report you provide (hability to clearly explain your approach, the methods used and the evolution of your prediction score), the codalab leader board score **WILL NOT** be used as a metric to evaluate your score. 








# IV. Few words about the dataset 

We dispose sample from many tissues, the idea is to evaluate if *tissue* could be a good predictor for genes.


**Try** ANOVA to predict missing values.


```{r}
sapply(gs, function(g){
  boxplot(d[[g]] ~ d$tissue, las=2, main=paste(g, "~ tissue"))  
})
```

```{r echo=TRUE, results="verbatim"}
pred = sapply(gs, function(g, d){
  m = lm(d[[g]] ~ d$tissue)
  predict(m, d)
}, d)


saveRDS(pred, "results.rds")
zip_filename = paste0("results_", format(Sys.time(), format="%m_%d_%Y_%s"), ".zip")
zip(zip_filename, "results.rds")
print(zip_filename)
```

```{r, code = readLines("scoring_program/scoring.r"), results="verbatim"}
```







# V. Submission, a detailed example

Produced results will be submitted to the "Linear Model Challenge" in the *Participate* tab.

https://competitions.codalab.org/competitions/21774#participate

How to submit results?
Produced results will be submitted to the “Linear Model Challenge” in the Participate tab.








# VI. References 


[NCBI/GENE] https://www.ncbi.nlm.nih.gov/gene/55227

[TCGA] Data are coming from the TCGA (The cancer genome atlas). 
They were extracted from several cohorts of patients. 
Here are TCGA Study Abbreviations corresponding to each cancer. 
https://gdc.cancer.gov/resources-tcga-users/tcga-code-tables/tcga-study-abbreviations

[Codalab] https://competitions.codalab.org







# VII. Session Information

```{r, results="verbatim"}
sessionInfo()
```


